### PR TITLE
Added events to zones

### PR DESCRIPTION
**Pull request by Njol**

### CODE REVIEW

**Comment by Njol** [comment]
Note: there's still an issue with interacting with chests on stage

**Comment by Njol** [comment]
Note that if this PR is not merged, a few mech changes need to be undone:
/home/epic/project_epic/server_config/data/scriptedquests/interactables/verdant/: re-enable these files
/home/epic/project_epic/server_config/data/datapacks/base/data/monumenta/advancements/trigger/generates_container_loot/verdant.json: re-enable
/home/epic/project_epic/server_config/data/scriptedquests/zone_layers/verdant/default.json: remove the last zone "Other"

File: plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java [hunk 1]
```diff
@@ -81,4 +79,8 @@ private void changeEvent(Player player, String layerName, String name) {
 
 		entry.changeEvent(mPlugin, player);
 	}
+
+	public Map<String, Map<String, ZoneProperty>> getZoneProperties() {
+		return mZoneProperties;
```
Comment (NickNackGus) [inline plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java:84]: This should create a new map of maps so the original is "immutable" (unless there's a better way to pull that off)

File: plugin/src/main/java/com/playmonumenta/scriptedquests/zones/event/ZoneEvent.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.playmonumenta.scriptedquests.zones.event;
+
+import com.playmonumenta.scriptedquests.utils.NmsUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+
+public abstract class ZoneEvent {
```
Comment (NickNackGus) [inline plugin/src/main/java/com/playmonumenta/scriptedquests/zones/event/ZoneEvent.java:9]: There's already two zone events with more descriptive names; this sounds like it's a base class for those and should be renamed.

File: plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java (hunk 1 above)
Comment (Combustible) [reply plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java:84]: Collections.UnmodifiableMap() maybe?

File: plugin/src/main/java/com/playmonumenta/scriptedquests/zones/event/ZoneEvent.java (hunk 2 above)
Comment (Combustible) [reply plugin/src/main/java/com/playmonumenta/scriptedquests/zones/event/ZoneEvent.java:9]: Named correctly, will leave

File: plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java (hunk 1 above)
Comment (Combustible) [reply plugin/src/main/java/com/playmonumenta/scriptedquests/managers/ZonePropertyManager.java]: Added that

**Pull request merged**